### ISSUE TITLE
Support Intent interception for all links

### ISSUE THREAD

**Issue by yashpalgoyal1304** [TeamNewPipe/NewPipe#7456]
### Checklist

- [x] I checked, but didn't find any duplicates (open OR closed) of this issue in the repo. 
issue 🔍 [newpipe as browser](https://github.com/TeamNewPipe/NewPipe/issues?q=is%3Aissue+is%3Aopen+newpipe+as+browser)
- [x] I have read the contribution guidelines given at https://github.com/TeamNewPipe/NewPipe/blob/HEAD/.github/CONTRIBUTING.md.
- [x] This issue contains only one feature request. I will open one issue for every feature I want to request.

#### What feature do you want?

* Newpipe may OPTIONALLY add separate activity for opening all `intent` links (think: like browser app)
* and check the incoming link with the newpipe's "supported links" list
* if the link matches/is supported by newpipe extractor, then open it as media/whatever
* otherwise pass it back to android system
* the whole mechanism being similar to open links with app
* i cant think of a suitable name for the service, but it can be smth like "Newpipe check links" ?!

#### Why do you want this feature?

+ This feature will add much more flexibility in the way links can be opened with newpipe. so, issues like:
    * https://github.com/TeamNewPipe/NewPipeExtractor/issues/746
    * https://github.com/TeamNewPipe/NewPipe/issues/7455
    * https://github.com/TeamNewPipe/NewPipe/issues/7436
+ The feature will add opportunity for newpipe extractor's "beta-support" services to be mass tested as well

Think of it like "dynamic intents" where the supported intents can be updated dynamically by the app. but this is not supported in android (and shouldnt be either imo).

#### Additional Context:
* I recently stumbled upon ability to add instance links in `Peertube instances` < `Content` < `Settings` ( [NewPipeExtractor#746] )
* even after adding link there, it was not supported by newpipe when i clicked such a link from other app
* .... since there can be so many links supported by newpipe, all those obv cant be added in the newpipe's intent
* but this restricts the ability to use newpipe to its full potential
* this proposal will make opening all these kinds of links in newpipe very very easy without causing any significant side effects

[NewPipeExtractor#746]: https://github.com/TeamNewPipe/NewPipeExtractor/issues/746

**Comment by mhmdanas** [comment]
Surely there will be some people who will have privacy concerns with an app having access to every single link they open, even if it's NewPipe.

Also, even if people have NewPipe installed, that does not mean they want e.g. YouTube links to open in NewPipe.

Do you have any solutions to the above problems?

**Comment by yashpalgoyal1304** [comment]
yeah, glad u asked it. i had smth in mind, but couldnt figure out with which context to say it.

So, with optional service, i meant that this will be additional service to the intents parsed in current manner. and there'd be a toggle in newpipe to disable that, saying smth like _Enable Newpipe link service_ with some description. so, users will have option to disable this additional service for once and for all

for the fine tuning control per link, i thought that there can be option: support all links, or support links in list only. the list can be proper syntax bounded list (maybe text editable?: 🤔). See the `Peertube instances` list for some starting inspiration.

**Comment by yashpalgoyal1304** [comment]
the user flow will be smth like following:

1. user clicks some link to a media service
2. they see the android's open with popup
3. if they want to try opening this media link with newpipe, THEN ONLY they will choose this service (say "newpipe browser")
4. that link will be compared, 
5. if it matches, then will be opened with newpipe
6. else will be redirected back to android sys

so, for addressing privacy concerns, there are several steps....
1. this service can be disabled
2. this is not at all intended to be default browser (a warning may be shown in this regard somehow) so,
3. ..... so, this will read link ONLY when user EXPLICITLY chooses newpipe_browser service

for addressing per service link choice, i have said above

**Comment by yashpalgoyal1304** [comment]
i got a bit verbose, sorry for that. but succinting things take a huuuuuuge chunk of mental efforts (and time) for me. so, couldnt do it atm 😅

**Comment by yashpalgoyal1304** [comment]
> * otherwise pass it back to android system

guess what, newpipe already has such a screen. tried clicking this link (from telegram):

* https://daily.bandcamp.com/album-of-the-day

<img src="https://user-images.githubusercontent.com/19423063/145017635-3b37e0a1-3563-4b89-9a3b-5700c0fcaf42.png" height=240px>
